### PR TITLE
Add doc examples on pointer types

### LINKED ISSUE #29336: API docs: raw pointers
Part of https://github.com/rust-lang/rust/issues/29329

http://doc.rust-lang.org/std/primitive.pointer.html

An important part of Rust's systems-level position, even if they're not safe to dereference.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @steveklabnik 

Fixes #29336

### CODE REVIEW

File: src/libcore/ptr.rs [hunk 1]
```diff
@@ -212,6 +212,14 @@ pub unsafe fn write_volatile<T>(dst: *mut T, src: T) {
 #[lang = "const_ptr"]
 impl<T: ?Sized> *const T {
     /// Returns true if the pointer is null.
+    ///
+    /// Example:
```
Comment (steveklabnik) [inline src/libcore/ptr.rs]: Can these be the `# Examples` headers like the rest of the docs?

File: src/libcore/ptr.rs [hunk 2]
```diff
@@ -212,6 +212,14 @@ pub unsafe fn write_volatile<T>(dst: *mut T, src: T) {
...
+    ///
```
Comment (steveklabnik) [inline src/libcore/ptr.rs]: Can we have a `Basic usage:` here, to match the rest of the new docs?

**Comment by steveklabnik** [comment]
Just some formatting things, I only left one comment, but it applies to all of them. Happy with this otherwise!

File: src/libcore/ptr.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/libcore/ptr.rs]: So I replace "Examples" by "#Current usage:"?

Comment (steveklabnik) [inline src/libcore/ptr.rs]: See http://doc.rust-lang.org/std/string/struct.String.html#method.new for an example.

(aaaand hopefully I can just point you at those guidelines I want to work on in the next few hours :smile: )

Comment (GuillaumeGomez) [inline src/libcore/ptr.rs]: Ooooooh. Isn't it a little overloaded to have "Example" and "basic usage" for just one example?

Comment (steveklabnik) [inline src/libcore/ptr.rs]: It is, but I currently value consistency over a bit of verbosity for simple examples.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by steveklabnik** [comment]
It's `Examples` with an `s`. Even though there's only one, same deal: consistency everywhere.

After that, this is finally good to go :smile:

**Comment by GuillaumeGomez** [comment]
And re-updated (I added ignore on `as_ref` examples to avoid failure since I think it's better to avoid using feature in examples).

**Pull request merged**